<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>702</width>
    <height>607</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Thermal Control System (UI Load Ver.)</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="PlotWidget" name="temp_plot_widget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>2</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item>
     <widget class="PlotWidget" name="pwm_plot_widget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="control_panel">
      <property name="topMargin">
       <number>10</number>
      </property>
      <item>
       <widget class="QGroupBox" name="grp_status">
        <property name="title">
         <string>Current Status</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label_static_temp">
           <property name="text">
            <string>Temperature:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="lbl_temp">
           <property name="styleSheet">
            <string>font-size: 30px; font-weight: bold; color: #2196F3;</string>
           </property>
           <property name="text">
            <string>0.0 Â°C</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="QLabel" name="lbl_fan">
           <property name="styleSheet">
            <string>font-size: 20px; font-weight: bold; color: gray;</string>
           </property>
           <property name="text">
            <string>FAN: OFF</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="2">
          <widget class="QLabel" name="lbl_pwm">
           <property name="styleSheet">
            <string>font-size: 20px;</string>
           </property>
           <property name="text">
            <string>PWM: 0%</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="grp_control">
        <property name="title">
         <string>Control Settings</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_control">
         <item>
          <widget class="QRadioButton" name="radio_manual">
           <property name="text">
            <string>Manual Mode</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radio_pid">
           <property name="text">
            <string>PID Mode</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget_manual" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_manual">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label_pwm_in">
              <property name="text">
               <string>PWM (%):</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spin_pwm">
              <property name="maximum">
               <number>100</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="chk_fan">
              <property name="text">
               <string>FAN ON</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget_pid" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_pid">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label_target">
              <property name="text">
               <string>Target Temp:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="spin_target">
              <property name="singleStep">
               <double>0.500000000000000</double>
              </property>
              <property name="value">
               <double>25.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="btn_apply">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>40</height>
            </size>
           </property>
           <property name="text">
            <string>Apply Settings</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_start">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="styleSheet">
            <string>background-color: #4CAF50; color: white; font-weight: bold; border-radius: 5px;</string>
           </property>
           <property name="text">
            <string>START SYSTEM</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_stop">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="styleSheet">
            <string>background-color: red; color: white; font-weight: bold; border-radius: 5px;</string>
           </property>
           <property name="text">
            <string>STOP (Emergency)</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="grp_save">
        <property name="maximumSize">
         <size>
          <width>600</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Data Recording</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_save">
         <item>
          <widget class="QLabel" name="label_fname">
           <property name="text">
            <string>File Name:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="edit_filename">
           <property name="placeholderText">
            <string>Empty = Timestamp</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_save">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>60</height>
            </size>
           </property>
           <property name="styleSheet">
            <string>background-color: #FF9800; color: white; font-weight: bold; border-radius: 5px;</string>
           </property>
           <property name="text">
            <string>SAVE DATA
(CSV + PNG)</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QGraphicsView</extends>
   <header>pyqtgraph</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
